src/components/NumericStepper/__tests__/NumericStepper.spec.js

```diff
@@ -0,0 +1,153 @@
+        wrapper.update();
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(0);
+        expect(wrapper.find('input').prop('value')).toEqual('0');
+    });
+    it('should react on stepUp click', () => {
+        wrapper = mount(<NumericStepper onChange={onChangeMock} step={2} />);
+
+        // CASE: Increase by 2 (to equal to 2) and make sure that number of onChange calls equal to 1
+        wrapper.find('StepperButton').at(1).prop('onClick')();
+        wrapper.update();
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(onChangeMock).toBeCalledWith(2);
+        expect(wrapper.find('input').prop('value')).toEqual('2');
+
+        // CASE: Increase by 2 (to equal to 4) and make sure that number of onChange calls equal to 2
+        wrapper.find('StepperButton').at(1).prop('onClick')();
+        wrapper.update();
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(4);
+        expect(wrapper.find('input').prop('value')).toEqual('4');
+    });
+    it('should disable buttons when limits are reached', () => {
+        wrapper = mount(
+            <NumericStepper onChange={onChangeMock} minValue={1} maxValue={3} defaultValue={2} />
+        );
+
+        // Simulate two clicks on stepUp button
+        wrapper.find('StepperButton').at(1).prop('onClick')();
+        wrapper.update();
+
+        expect(wrapper.find('input').prop('value')).toEqual('3');
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(wrapper.find('StepperButton').at(1).prop('disabled')).toBe(true);
+
+        // Simulate two clicks on stepDown button
+        wrapper.find('StepperButton').at(0).prop('onClick')();
+        expect(onChangeMock).toBeCalledTimes(2);
+        wrapper.update();
+        wrapper.find('StepperButton').at(0).prop('onClick')();
+        expect(onChangeMock).toBeCalledTimes(3);
+        wrapper.update();
+
+        expect(wrapper.find('input').prop('value')).toEqual('1');
+        expect(wrapper.find('StepperButton').at(0).prop('disabled')).toBe(true);
+    });
+    it('should set bottom edge value if user enters value below allowed limit', () => {
+        wrapper = mount(<NumericStepper onChange={onChangeMock} minValue={10} />);
+        const data = { target: { value: '9' } };
+
+        wrapper.find('input').simulate('change', data);
+        wrapper.find('input').simulate('blur');
+
+        expect(wrapper.find('input').prop('value')).toEqual('10');
+    });
+    it('should set top edge value if user enters value above allowed limit', () => {
+        wrapper = mount(<NumericStepper onChange={onChangeMock} maxValue={10} />);
+        const data = { target: { value: '11' } };
+
+        wrapper.find('input').simulate('change', data);
+        wrapper.find('input').simulate('blur');
+
+        expect(wrapper.find('input').prop('value')).toEqual('10');
+    });
+    it('should react on keyup/keydown press', () => {
```

Comment:
I tried many ways to do this, but does anyone have the idea that might work?

Reply:
I also tried several approaches, but it doesn't realy help. I think we need to create a follow-up story for investigating the issue.

Reply:
Created ticket for that: https://jira.textkernel.nl/browse/ONEUI-280